pandas/io/formats/format.py

```diff
@@ -1860,7 +1860,7 @@ def _format_strings(self):
             (lambda x: pprint_thing(x, escape_chars=('\t', '\r', '\n'))))
 
         def _format(x):
-            if self.na_rep is not None and lib.checknull(x):
+            if self.na_rep is not None and libmissing.checknull(x):
```

Comment:
in followup (or here), change this to

``is_scalar(x) and isna(x)``; this is reaching too much into the internals.

Reply:
and all below here

*maybe* worth defining a ``is_scalar_na`` function